Source/Core/Core/Boot/Boot_WiiWAD.cpp

```diff
@@ -21,17 +23,11 @@
 #include "DiscIO/Volume.h"
 #include "DiscIO/VolumeCreator.h"
 
-static u32 state_checksum(u32* buf, int len)
+static u32 StateChecksum(const u32* buffer, size_t length)
 {
-  u32 checksum = 0;
-  len = len >> 2;
+  length >>= 2;
```

Comment:
I'm assuming this is because `length` isn't the array length of `buffer`, but its memory size in bytes. Maybe the name could use some adjustment to make that clear; or perhaps change it to pass in the array size instead?
Also, if it sticks around, can you make it a "divide by 4" (or divide by `sizeof(u32)`) rather than the shift? Compilers should be able to optimize this anyways, with ourselves as the ones writing the code focusing on readability instead (if possible).